nomad/structs/job.go

```diff
@@ -144,3 +144,21 @@ func (j *Job) RequiredBridgeNetwork() set.Collection[string] {
 	}
 	return result
 }
+
+// RequiredTransparentProxy identifies which task groups, if any, within the job
+// contain Connect blocks using transparent proxy
+func (j *Job) RequiredTransparentProxy() set.Collection[string] {
+	result := set.New[string](len(j.TaskGroups))
+	for _, tg := range j.TaskGroups {
+		for _, service := range tg.Services {
+			if service.Connect != nil {
+				if service.Connect.HasTransparentProxy() {
+					result.Insert(tg.Name)
+					continue
```

Comment:
is this `continue` future-oriented?  as I read it here, it seems like a no-op, since nothing else follows it.

perhaps a `break` instead, to pop out of the inner `Services` loop, since any single service with tproxy enables it for the whole group/alloc, right?

Reply:
Fixed, and added a unit test anyways :grinning: